### PR TITLE
BUOYS: Add 2d-3d multiobservation pose estimator

### PR DESCRIPTION
**Pull request by jpanikulam**
Super simple, estimate pose as the best intersection of N rays
- Also makes buoy mission work with the new stuff
- Also adds use_sim_time to gazebo launch
- Also fixes crappy rotation control in the simulator

### CODE REVIEW

File: command/sub8_missions/missions/buoy.py [hunk 1]
```diff
@@ -1,89 +1,55 @@
...
-        return
+        defer.returnValue(False)
 
-    yield sub.move.height(full_transform._p[2]).go(speed=0.2)
+    yield sub.move.depth(-full_transform._p[2]).go(speed=0.2)
     yield util.wall_sleep(0.2)
     print 'looking at'
-    yield sub.move.look_at(full_transform._p).go(speed=0.1)
-    print 'position'
+    yield sub.move.look_at_without_pitching(full_transform._p).go()
```
Comment (mattlangford) [inline command/sub8_missions/missions/buoy.py:35]: Why do you do this 3 times?

File: gnc/sub8_perception/nodes/buoy_2d.py [hunk 2]
```diff
@@ -165,30 +201,24 @@ def find_single_buoy(self, img, buoy_type):
...
-                    for particle in self.ppf.particles[:, ::50].transpose():
-                        k += 1
-                        self.rviz.draw_sphere(
-                            particle,
-                            color=(0.8, 0.2, 0.0, 0.7),
-                            scaling=(0.05, 0.05, 0.05),
-                            _id=k,
-                            frame='/map'
-                        )
+            (t, rot_q) = self.transformer.lookupTransform('/map', '/stereo_front/right', self.last_image_time - rospy.Duration(0.04))
```
Comment (mattlangford) [inline gnc/sub8_perception/nodes/buoy_2d.py:206]: Is this to fix the camera stamped in the future problem? Regardless @rleyva should know about it if you haven't told him.

File: simulation/sub8_gazebo/diagnostics/gazebo_tests/test_buoy_bump.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    def run(self, sub):
+        print 'running'
+        # yield sub.move.right(2.0).go()
+        # yield sub.move.down(1.0).go()
+        # yield sub.move.left(2.0).go()
+        # yield sub.move.up(1.0).go()
+
+        yield buoy.run(sub)
+
+        success = True
```
Comment (mattlangford) [inline simulation/sub8_gazebo/diagnostics/gazebo_tests/test_buoy_bump.py:38]: Did you want to add collision detection to the Job class in this PR?
If not I will make sure to do it for my next one.

File: command/sub8_missions/missions/buoy.py [hunk 4]
```diff
@@ -1,89 +1,55 @@
...
+    yield util.wall_sleep(0.2)
+
+    dist = np.inf
+    while(dist > 1.0):
+        dist = np.linalg.norm(full_transform._p - sub.pose.position) * 0.5
+        yield sub.move.look_at_without_pitching(full_transform._p).go()
+        yield util.wall_sleep(0.2)
+
+        yield sub.move.forward(dist).go()
```
Comment (mattlangford) [inline command/sub8_missions/missions/buoy.py:50]: Why not just move right up next to it the estimated position in one fell swoop here? 
It seems like multiple iterations would only help if we re-estimated the position of the bouy each time we moved.

File: simulation/sub8_gazebo/diagnostics/gazebo_tests/test_buoy_bump.py (hunk 3 above)
Comment (jpanikulam) [inline simulation/sub8_gazebo/diagnostics/gazebo_tests/test_buoy_bump.py:38]: Counting on you my man!

File: gnc/sub8_perception/nodes/buoy_2d.py (hunk 2 above)
Comment (jpanikulam) [inline gnc/sub8_perception/nodes/buoy_2d.py:206]: This is -- except it even happens in simulation. wtf? What's the point of camera time stamps if they are not real?

File: command/sub8_missions/missions/buoy.py (hunk 1 above)
Comment (jpanikulam) [inline command/sub8_missions/missions/buoy.py:35]: To persuade the sim to work, in hindsight, not needed

File: gnc/sub8_perception/nodes/buoy_2d.py (hunk 2 above)
Comment (mattlangford) [inline gnc/sub8_perception/nodes/buoy_2d.py:206]: Why do we have mirrors if our eyes aren't real?

File: simulation/sub8_gazebo/diagnostics/gazebo_tests/test_buoy_bump.py (hunk 3 above)
Comment (mattlangford) [inline simulation/sub8_gazebo/diagnostics/gazebo_tests/test_buoy_bump.py:38]: Okay!

**Comment by jpanikulam** [comment]
I need to grab some of the edits I made on the sub this morning

**Pull request merged**